Keas.Mvc/ClientApp/components/Workstations/WorkstationDetails.tsx

```diff
@@ -16,73 +17,73 @@ interface IProps {
   updateSelectedWorkstation: (workstation: IWorkstation, id?: number) => void;
 }
 
-export default class WorkstationDetails extends React.Component<IProps, {}> {
-  public static contextType = Context;
-  public context!: React.ContextType<typeof Context>;
+const WorkstationDetails = (props: IProps) => {
+  const context = useContext(Context);
+  const workstation = props.selectedWorkstation;
 
-  public componentDidMount() {
-    if (!this.props.selectedWorkstation) {
+  useEffect(() => {
```

Comment:
Are you sure you want this to run on every render?  I think maybe instead did you mean to add the empty dependency array `useEffect(()=>, [])` and have it run just once?  I feel like otherwise it's going to run over and over in a loop as the fetch method causes the parent props to change.  Maybe not a huge issue, but I think it's good practice to always provide the second argument to useEffect()

Reply:
Shoot forgot about that. I will change that right now.